common/key_value_parser/key_value_parser.cpp

```diff
@@ -60,22 +69,23 @@ ExpectedKeyValuesMap ParseKeyValues(const vector<string> &items, char delimiter)
 
 		string key = str.substr(0, delim_pos);
 		string value = str.substr(delim_pos + 1, str.size() - delim_pos - 1);
-		if (ret.count(key) != 0) {
-			ret[key].push_back(value);
+		if (base.count(key) != 0) {
+			base[key].push_back(value);
 		} else {
-			ret[key] = {value};
+			base[key] = {value};
 		}
 	}
 
 	if (invalid_data != "") {
-		return ExpectedKeyValuesMap(MakeError(
+		return MakeError(
 			KeyValueParserErrorCode::InvalidDataError,
-			"Invalid data given: '" + invalid_data + "'"));
+			"Invalid data given: '" + invalid_data + "'");
 	}
 
-	return ExpectedKeyValuesMap(ret);
+	return MakeError(KeyValueParserErrorCode::NoError, "");
```

Comment:
Q: Should we consider having an overload with just the error, and then we won't have to pass in an empty message?

Reply:
Hmm, that would allow creation of errors with other codes and no message. And that we don't want.

Reply:
You should return `error::NoError` instead. The `NoError`s for other error categories need to be there for code correctness purposes, but there is no need to use them.